src/main/java/uk/gov/companieshouse/web/lfp/service/navigation/NavigatorService.java

```diff
@@ -62,105 +60,78 @@ private Class getControllerClass(Class clazz, Direction direction, String ... pa
      * @param  clazz the controller class in the chain to begin the scan at
      * @return the next controller class in the chain
      */
-    private Class getNextControllerClass(Class clazz, String ... pathVars) {
-        Annotation nextControllerAnnotation = AnnotationUtils.findAnnotation(clazz, NextController.class);
+    private Class<?> getNextControllerClass(Class<?> clazz, String ... pathVars) {
+        NextController nextControllerAnnotation = AnnotationUtils.findAnnotation(clazz, NextController.class);
         if (nextControllerAnnotation == null) {
-            throw new MissingAnnotationException("Missing @NextController annotation on " + clazz.toString());
+            throw new MissingAnnotationException("Missing @NextController annotation on " + clazz);
         }
 
-        Class[] classList = ((NextController) nextControllerAnnotation).value();
```

Comment:
Duplicated lines are now refactored into a method and complexity ratio is also reduced